### PR TITLE
Adding path parameters and disposing additive context in both frameworks

### PR DESCRIPTION
**Pull request by anna-git**
Fixes #

Changes proposed in this pull request:
- disposing AdditiveContext per request
- passing path parameters to the waf when they are available

@DataDog/apm-dotnet

### CODE REVIEW

File: src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs [hunk 1]
```diff
@@ -665,16 +651,14 @@ private void OnHostingHttpRequestInStart(object arg)
...
-                { "server.request.method", request.Method },
-                { "server.request.uri.raw", url },
-                { "server.request.query", request.QueryString.ToString() },
-            };
+            var dic = request.PrepareArgsForWaf();
+            security.InstrumentationGateway.RaiseEvent(dic, new HttpTransport(context));
+        }
 
-            security.InstrumentationGateway.RaiseEvent(dict, new HttpTransport(context));
+        private void RaiseIntrumentationEvent(IDatadogSecurity security, HttpContext context, HttpRequest request, RouteData routeDatas)
```
Comment (robertpi) [inline src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs:658]: I think you could merge the two `RaiseIntrumentationEvent` by make `routeDatas` optional.

('Data' is uncountable in English, so always singular)

File: devenv.bat [hunk 2]
```diff
@@ -53,9 +53,10 @@ rem Enable .NET Core Profiling API
 SET CORECLR_ENABLE_PROFILING=1
 SET CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
 SET CORECLR_PROFILER_PATH=%~dp0\src\Datadog.Trace.ClrProfiler.Native\bin\%profiler_configuration%\%profiler_platform%\Datadog.Trace.ClrProfiler.Native.dll
+set DD_TRACE_CALLTARGET_ENABLED=0
 
 rem Enable Security features
-SET DD_ENABLE_SECURITY=true
+SET DD_ENABLE_SECURITY=1
```
Comment (robertpi) [inline devenv.bat:59]: Doesn't true work? I think it should, otherwise we should correct the code that parses it.

**Review by robertpi — APPROVED** [review]
Looks good to me.

File: src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AspNet/AsyncControllerActionInvoker_BeginInvokeAction_Integration.cs [hunk 3]
```diff
@@ -56,8 +58,15 @@ public class AsyncControllerActionInvoker_BeginInvokeAction_Integration
             {
                 if (HttpContext.Current != null)
                 {
-                    scope = AspNetMvcIntegration.CreateScope(controllerContext.DuckCast<ControllerContextStruct>());
+                    var duckedControllerContext = controllerContext.DuckCast<ControllerContextStruct>();
+                    HttpContext.Current.Request.PrepareArgsForWaf(duckedControllerContext.RouteData);
```
Comment (kevingosse) [inline src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AspNet/AsyncControllerActionInvoker_BeginInvokeAction_Integration.cs:62]: It doesn't look like you're using the return value? What is the purpose of this call?

Comment (anna-git) [reply src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AspNet/AsyncControllerActionInvoker_BeginInvokeAction_Integration.cs]: yes thank you!

**Pull request merged**